core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsAddressEndpointGroup.java

```diff
@@ -105,7 +106,7 @@ public static DnsAddressEndpointGroupBuilder builder(String hostname) {
             String hostname, @Nullable ResolvedAddressTypes resolvedAddressTypes) {
 
         if (resolvedAddressTypes == null) {
-            if (NetUtil.isIpV4StackPreferred()) {
+            if (NetUtil.isIpV4StackPreferred() || !anyInterfaceSupportsIpV6()) {
```

Comment:
I think this code has been around for a long time - does it mean it's not a regression? It seems like it used to work though, but possibly an interaction with something else rather than a regression. Just want to check.

Reply:
Yeah, I might make this too impromptu. Let me dig this more tomorrow. 😅

Reply:
@anuraaga, it now skips link-local addresses as well. Shall we merge, release and see if the problem goes away?

Reply:
Yeah let's try